Common/src/geometry/primal_grid/CHexahedron.cpp

```diff
@@ -26,47 +26,32 @@
  */
 
 #include "../../../include/geometry/primal_grid/CHexahedron.hpp"
+#include "../../../include/option_structure.hpp"
 
-unsigned short CHexahedron::Faces[6][4] = {{0,1,5,4},{1,2,6,5},{2,3,7,6},{3,0,4,7},{0,3,2,1},{4,5,6,7}};
-
-unsigned short CHexahedron::Neighbor_Nodes[8][3] = {{1,3,4},{0,2,5},{1,3,6},{0,2,7},{0,5,7},{4,6,1},{2,5,7},{4,3,6}};
-
-unsigned short CHexahedron::nNodesFace[6] = {4,4,4,4,4,4};
-
-unsigned short CHexahedron::nNeighbor_Nodes[8] = {3,3,3,3,3,3,3,3};
-
-unsigned short CHexahedron::nFaces = 6;
-
-unsigned short CHexahedron::nNodes = 8;
-
-unsigned short CHexahedron::nNeighbor_Elements = 6;
-
-unsigned short CHexahedron::VTK_Type = 12;
-
-unsigned short CHexahedron::maxNodesFace = 4;
+constexpr unsigned short CHexahedronConnectivity::nNodes;
+constexpr unsigned short CHexahedronConnectivity::nFaces;
+constexpr unsigned short CHexahedronConnectivity::nNodesFace[6];
+constexpr unsigned short CHexahedronConnectivity::maxNodesFace;
+constexpr unsigned short CHexahedronConnectivity::Faces[6][4];
+constexpr unsigned short CHexahedronConnectivity::nNeighbor_Nodes[8];
+constexpr unsigned short CHexahedronConnectivity::Neighbor_Nodes[8][3];
+constexpr unsigned short CHexahedronConnectivity::VTK_Type;
 
 CHexahedron::CHexahedron(unsigned long val_point_0, unsigned long val_point_1,
              unsigned long val_point_2, unsigned long val_point_3,
              unsigned long val_point_4, unsigned long val_point_5,
-             unsigned long val_point_6, unsigned long val_point_7) : CPrimalGrid() {
-  unsigned short iNeighbor_Elements;
-
-  nDim = 3;
-
+             unsigned long val_point_6, unsigned long val_point_7):
+  CPrimalGridWithConnectivity<CHexahedronConnectivity>(false)
+{
   /*--- Allocate and define face structure of the element ---*/
-  Nodes = new unsigned long[nNodes];
+  Nodes.resize(GetnNodes());
```

Comment:
The member function `GetNodes()` 
- is declared virtual in `CPrimalGrid`, 
- has a final override in the derived class `CPrimalGridWithConnectivity<CHexahedronConnectivity>`, and 
- is used in the constructor of the derived class `CHexahedron`.

I know that virtual functions cannot be used in the constructor as freely as afterwards, because base classes are constructed before derived classes. However in this case, I think it should be fine because the virtual version of `GetnNodes`, that we want to use in the constructor, is defined in a base class (and moreover, is marked final there).

Reply:
It's fine, you can call the virtual members of parent classes, if they are final there is no chance of errors.